gensim/models/atmodel.py

```diff
@@ -0,0 +1,1017 @@
+        author2doc and doc2author corresponding to this test set are provided. It is
+        not recommended to call this method with data that has authors the model has 
+        not seen; if this is the case, those documents will simply be discarded.
+
+        To obtain the per-word bound, compute:
+        >>> corpus_words = sum(cnt for document in corpus for _, cnt in document)
+        >>> model.bound(corpus, author2doc=author2doc, doc2author=doc2author) / corpus_words
+
+        """
+
+        # NOTE: it may be possible to enable evaluation of documents with new authors. To
+        # do this, self.inference() has to be altered so that it uses gamma = self.state.gamma[a, :]
+        # if author a is already trained on, but initializes gamma randomly if author a is
+        # not already in the model.
+
+        _lambda = self.state.get_lambda()
+        Elogbeta = dirichlet_expectation(_lambda)
+        expElogbeta = np.exp(dirichlet_expectation(_lambda))
+
+        if author2doc is None and doc2author is None:
+            gamma = self.state.gamma
+            chunk_idx = [d for d in xrange(len(chunk))]
+            author2doc = self.author2doc
+            doc2author = self.doc2author
+        else:
+            # Infer gamma based on input corpus.
+
+            # Will be needed in self.inference().
+            def rho():
+                return pow(self.offset + self.passes + (self.num_updates / self.chunksize), -self.decay)
+
+            # sstats are not collected, thus lambda is not updated
+            gamma, _ = self.inference(chunk, author2doc, doc2author, rho())
+
+            # Bound of held-out (test) data can only be computed with authors
+            # that are already existing in the data.
+            # Documents that contain new authors are discarded.
+            num_docs_new_authors = 0
+            chunk_idx = []
+            for d in xrange(len(chunk)):
+                authors_d = doc2author[d]
+                doc_new_authors = False
+                for a in authors_d:
+                    if not self.author2doc.get(a):
+                        doc_new_authors = True
+                if not doc_new_authors:
+                    chunk_idx.append(d)
+                else:
+                    num_docs_new_authors += 1
+            if num_docs_new_authors > 0:
+                logger.warning('bound() called with held-out data with new authors; discarding %d documents.' % (num_docs_new_authors))
+
+
+        Elogtheta = dirichlet_expectation(gamma)
+        expElogtheta = np.exp(dirichlet_expectation(gamma))
+
+        word_score = 0.0
+        authors_set = set()  # Used in computing theta bound.
+        theta_score = 0.0
+        for d in chunk_idx:
+            if author2doc is None:
+                doc_no = chunk_doc_idx[d]
+            else:
+                doc_no = d
+            doc = chunk[d]
+            authors_d = doc2author[doc_no]
+            authors_d = [self.author2id[a] for a in authors_d]
```

Comment:
Changed it to `authors_d = [self.author2id[a] for a in doc2author[doc_no]]`. It does look better.